### PR TITLE
Streamline investment income calculations

### LINKED ISSUE #1845: Confusing specification of EITC investment income
EITC investment income is currently defined in the [`eitc_eligible.py` formula](https://github.com/PolicyEngine/policyengine-us/blob/master/policyengine_us/variables/gov/irs/credits/earned_income/eitc_eligible.py) as a combination of a parameter list and formula calculations.  This is confusing because most people would expect an income sources list to be complete.

The reverse problem exists for the NIIT base: net investment income.  It is constructed in the [`net_investment_income_tax.py` formula](https://github.com/PolicyEngine/policyengine-us/blob/master/policyengine_us/variables/gov/irs/tax/federal_income/net_investment_income_tax.py) when it could be constructed with a simple `adds` statement.  And then this new `net_investment_income` variable could be used to streamline the calculation of EITC investment income.

### PR DESCRIPTION
**Pull request by martinholmer**
Fixes #1845.
These code-health changes cause no changes in the "make test" results.
They also leave unchanged all the validation test results reported in issue #993.

### CODE REVIEW

File: policyengine_us/variables/gov/irs/credits/earned_income/eitc_eligible.py [hunk 1]
```diff
@@ -17,18 +17,23 @@ def formula(tax_unit, period, parameters):
...
-            add(tax_unit, period, inv_income_sources)
-            + max_(0, tax_unit("c01000", period))
-            + max_(
-                0,
-                tax_unit("tax_unit_rental_income", period)
-                - tax_unit("tax_unit_partnership_s_corp_income", period),
-            )
+        no_loss_capital_gains = max_(
+            0,
+            add(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/irs/credits/earned_income/eitc_eligible.py:22]:
```suggestion
            tax_unit("capital_gains", period)
```

File: policyengine_us/variables/gov/irs/credits/earned_income/eitc_eligible.py [hunk 2]
```diff
@@ -17,18 +17,23 @@ def formula(tax_unit, period, parameters):
...
+                tax_unit,
+                period,
+                ["short_term_capital_gains", "long_term_capital_gains"],
+            ),
+        )
+        eitc_investment_income = (
+            tax_unit("net_investment_income", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/irs/credits/earned_income/eitc_eligible.py:29]:
```suggestion
            add(tax_unit, ["net_investment_income", "tax_exempt_interest_income"], period)
```

File: policyengine_us/variables/gov/irs/tax/federal_income/net_investment_income_tax.py [hunk 3]
```diff
@@ -10,16 +10,10 @@ class net_investment_income_tax(Variable):
...
-        NII_ELEMENTS = [
-            "taxable_interest_income",
-            "dividend_income",
-            "c01000",
-            "rental_income",
-        ]
-        nii = max_(0, add(tax_unit, period, NII_ELEMENTS))
         p = parameters(period).gov.irs.investment.net_investment_income_tax
         threshold = p.threshold[tax_unit("filing_status", period)]
         base = min_(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/irs/tax/federal_income/net_investment_income_tax.py:15]:
```suggestion
        excess_agi = max_(0, tax_unit("adjusted_gross_income", period) - threshold)
        base = min_(
```

File: policyengine_us/variables/gov/irs/tax/federal_income/net_investment_income_tax.py [hunk 4]
```diff
@@ -10,16 +10,10 @@ class net_investment_income_tax(Variable):
...
-            nii, max_(0, tax_unit("adjusted_gross_income", period) - threshold)
+            max_(0, tax_unit("net_investment_income", period)),
+            max_(0, tax_unit("adjusted_gross_income", period) - threshold),
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/irs/tax/federal_income/net_investment_income_tax.py:17]:
```suggestion
            excess_agi,
```

**Review by MaxGhenis — APPROVED** [review]
Thanks for this clean up, just a couple minor suggestions

File: policyengine_us/parameters/gov/irs/investment/income/sources.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+description: Income sources counted as net investment income.
+values:
+  2013-01-01:
+    - taxable_interest_income
+    - dividend_income
+    - rental_income
+    - c01000  # loss-limited capital gains
+metadata:
+  unit: currency-USD
+  name: irs_investment_income_sources
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/irs/investment/income/sources.yaml:10]:
```suggestion
```

File: policyengine_us/variables/gov/irs/credits/earned_income/eitc_eligible.py (hunk 1 above)
Comment (martinholmer) [reply policyengine_us/variables/gov/irs/credits/earned_income/eitc_eligible.py:22]: Done.

File: policyengine_us/variables/gov/irs/credits/earned_income/eitc_eligible.py (hunk 2 above)
Comment (martinholmer) [reply policyengine_us/variables/gov/irs/credits/earned_income/eitc_eligible.py:29]: Done.

File: policyengine_us/parameters/gov/irs/investment/income/sources.yaml (hunk 5 above)
Comment (martinholmer) [reply policyengine_us/parameters/gov/irs/investment/income/sources.yaml:10]: Done.

File: policyengine_us/variables/gov/irs/credits/earned_income/eitc_eligible.py (hunk 1 above)
Comment (martinholmer) [reply policyengine_us/variables/gov/irs/credits/earned_income/eitc_eligible.py:22]: When I implement your suggestion, I get 64 errors from "make test".  Here is a typical error:
```
policyengine_us/variables/gov/irs/credits/earned_income/eitc_eligible.py:22: in formula
    tax_unit("capital_gains", period),
//Users/mrh/anaconda3/envs/policyengine-us/lib/python3.9/site-packages/policyengine_core/populations/population.py:100: in __call__
    self.entity.check_variable_defined_for_entity(variable_name)
//Users/mrh/anaconda3/envs/policyengine-us/lib/python3.9/site-packages/policyengine_core/entities/entity.py:52: in check_variable_defined_for_entity
    raise ValueError(message)
E   ValueError: You tried to compute the variable 'capital_gains' for the entity 'tax_units';
E   however the variable 'capital_gains' is defined for 'people'.
```
I'm going to skip this suggested code change.

**Comment by martinholmer** [comment]
@MaxGhenis,  When I fix the test errors, I get this.
```
(policyengine-us) policyengine-us% git push origin inv-inc
To https://github.com/martinholmer/policyengine-us
 ! [rejected]            inv-inc -> inv-inc (fetch first)
error: failed to push some refs to 'https://github.com/martinholmer/policyengine-us'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. **This is usually caused by another repository pushing
hint: to the same ref.** You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```
So, you've hijack this PR it seems.  OK, fine.  You finish it up.

**Comment by MaxGhenis** [comment]
Sorry, I thought you were done with my comments so I just wanted to add a couple cleanups. Merging.

**Pull request merged**